### ISSUE TITLE
IN operator not working on `Short` type fields

### ISSUE THREAD

**Issue by mehdizarepour** [orientechnologies/orientdb#7578]
### OS: Linux, MacOS

I want to use `IN` operator to filter records.

## Steps to reproduce  
```sql
/* schema */
create class User extends V;
create property User.state Short;

/* data*/
create vertex User set state = 1;
create vertex User set state = 1;
create vertex User set state = 2;
```
queries:

```sql
select from User where state in [1]

select from User where state in 1
```

## Expected/Actual behavior  
These queries should return two records, but instead, it returns nothing.
It should be noted that when I use `Integer` instead of `Short`, the queries are work as expected.

Thanks

**Comment by luigidellaquila** [comment]
Hi @mehdizarepour 

Thank you for reporting, I'll check it asap

Thanks

Luigi

**Comment by mehdizarepour** [comment]
Hi @luigidellaquila 

The `CONTAINS` operator has the same problem.

**Comment by luigidellaquila** [comment]
Hi @mehdizarepour 

I just pushed a fix on 2.2.x, the fix will be released with 2.2.25

Thanks

Luigi
